CMakeLists.txt

```
@@ -324,6 +324,15 @@ if(WIN32 AND MINGW)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -static-libstdc++")
 endif()
 
+if(WIN32)
```

```suggestion
// Check if inet_pton is already available, to avoid conflicts with the implementation in LightGBM.
// As of 2022, MinGW started including a definition of inet_pton.
if(WIN32)
```

I think we should still preserve something like the comment that's currently in `socket_wrapper.hpp`. I think it's useful for explaining why this check is here.